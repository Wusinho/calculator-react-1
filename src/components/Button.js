import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ buttonName, clickHandler }) => {
  const handleClick = () => clickHandler(buttonName);
  return (
    <button type="button" onClick={handleClick}>{buttonName}</button>
  );
};

Button.propTypes = {
  buttonName: PropTypes.string,
  clickHandler: PropTypes.func.isRequired,
};

Button.defaultProps = {
  buttonName: null,
};

export default Button;
