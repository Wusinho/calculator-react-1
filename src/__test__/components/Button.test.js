import React from 'react';
import { create } from 'react-test-renderer';

import Button from '../../components/Button';

it('Should renders correctly', () => {
  const elem = create(<Button buttonName="button" clickHandler={() => 'this is a button'} />);
  expect(elem).toMatchSnapshot();
});
