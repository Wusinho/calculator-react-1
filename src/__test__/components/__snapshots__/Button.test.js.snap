// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should renders correctly 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  button
</button>
`;
